### DISCUSSION TITLE
Intermittent 5xx responses from the pulls API call

### DISCUSSION THREAD

**Discussion by Inglonias** [community/community#24673] [Apps, API and Webhooks]
<p>The API call to return a list of pull requests is returning 5xx status codes seemingly at random. The API status page reads green, so I wanted to see if anyone else is having this issue. I know at least <a href="https://github.com/qrrk/Catapult/pull/19#issuecomment-994993832" rel="noopener nofollow ugc">one other GitHub user has hit this problem</a></p>
<p><code>https://api.github.com/repos/cleverraven/cataclysm-dda/pulls?state=closed&amp;sort=created&amp;direction=desc&amp;per_page=100</code></p>

**Comment by Inglonias** [comment]
<p>It looks like making authenticated API calls mitigates this issue entirely.</p>
<p>EDIT: A new theory we had is that the Cataclysm-DDA repository is simply too large and has too many pull requests for the API to handle consistently.</p>

**Comment by Inglonias** [comment]
<p>I suspect what is happening here has to do with the rather large size of the Cataclysm DDA repo. It has over 33,000 closed pull requests, and going through a list of that size may be too much to handle for the servers tasked with handling unauthenticated GitHub API calls.</p>

**Comment by Inglonias** [comment, marked as answer]
<p>After a conversation with GitHub support, I have an answer. Because the repo is so large, unauthenticated requests are hitting a timeout cap because they take too long. The solution was to switch to the GitHub search API, searching for pull requests rather than listing pull requests directly.</p>

**Comment by jfullstackdev** [comment]
<p>I was thinking the same too, but I did not post at the very first place, because I might not be the appropriate one to answer,</p>
<p>the API is still limited particularly when being accessed by unauthenticated, so if that is a mix and the repo is so large, the server will think it’s security vulnerability</p>
